File: app/src/main/java/art/coded/wireframe/model/entity/ElementComparator.java
package art.coded.wireframe.model.entity;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public class ElementComparator extends DiffUtil.ItemCallback<Element> {

    @Override public boolean areItemsTheSame(@NonNull Element oldItem, @NonNull Element newItem) {
        return oldItem.hashCode() == newItem.hashCode();
    }

    @Override public boolean areContentsTheSame(@NonNull Element oldItem, @NonNull Element newItem) {
        return oldItem.getName().equals(newItem.getName());
    }
}

File: app/src/ktx/java/art/coded/wireframe/model/entity/ElementComparator.kt
package art.coded.wireframe.model.entity

import androidx.recyclerview.widget.DiffUtil

class ElementComparator : DiffUtil.ItemCallback<Element>() {
    override fun areItemsTheSame(oldItem: Element, newItem: Element): Boolean {
        return oldItem.hashCode() == newItem.hashCode()
    }

    override fun areContentsTheSame(oldItem: Element, newItem: Element): Boolean {
        return oldItem.name == newItem.name
    }
}